src/pwm.rs

```diff
@@ -258,10 +370,59 @@ macro_rules! hal {
                         .set_bit()
                 );
 
-                unsafe { mem::MaybeUninit::uninit().assume_init() }
+                Pwm {
+                    clk: clk,
+                    _tim: PhantomData,
+                    _channels: PhantomData
+                }
+            }
+            
+            pwm_impl!(
+                $TIMX, $timX, (0, 1, 2, 3), (C1, C2, C3, C4);
```

Comment:
Looks like you make 14 implementation for each combination of `PINS::Channels`. (14*14)
This is incorrect. Only 1 implementation must be here.
If you port this from somewhere, please add link.

Reply:
Not quite.  I do 14 combinations for each defined timer from the from `hal!` expansions.  So, in this case it would be 14*4.  The impl is required multiple times because of the concretely typed channel configurations.  I tried to do it outside with generics but it would not work.

Reply:
Also, I did not port any code from other sources.  I just tried to mimic the existing coding style in this file.  I did get some assistance in the Discord concerning macros.  But that was far from a copy-paste operation.

Reply:
All that being said, if there was a way to do the impl in there without the macro expansion, I agree, less code for the compiler, the better...

Reply:
Sorry, you are right. I thought that this is inside `fn $timX<REMAP, P, PINS>(`.